packages/eventual-send/src/index.js

```diff
@@ -304,42 +350,64 @@ export function makeHandledPromise() {
       rejectHandled,
       resolveWithPresence,
     );
+
     return handledP;
   }
 
-  HandledPromise.prototype = Promise.prototype;
-  Object.setPrototypeOf(HandledPromise, Promise);
-
   function isFrozenPromiseThen(p) {
     return (
       isFrozen(p) &&
       getPrototypeOf(p) === Promise.prototype &&
       Promise.resolve(p) === p &&
-      gopd(p, 'then') === undefined
+      getOwnPropertyDescriptor(p, 'then') === undefined
     );
   }
 
+  /** @type {import('.').HandledPromiseStaticMethods} */
   const staticMethods = {
-    get(target, key) {
-      return handle(target, 'get', key);
+    get(target, prop) {
+      // Coerce property to string or symbol.
+      if (typeof prop !== 'symbol') {
+        prop = String(prop);
+      }
```

Comment:
You do this a lot. Please create a reusable abstraction for it. If there is already a conventional name for it, use that.